File: 4__Ruby_Program/10__number_guessing_game.rb
# Number Guessing Game

puts "=== Number Guessing Game ==="

play = 10
win = false

random_number = rand(1..100)

for i in 1..play
    print "[Turn #{i} / #{play}] Choose random number from 0 - 100: "
    choose_number = gets.chomp.to_i
    
    if choose_number > random_number
        puts "The number is smaller than this!"
    elsif choose_number < random_number
        puts "The number is bigger than this!"
    else
        puts "You win! The number is #{random_number}"
        win = true
        break
    end
end

unless win
    puts "You lose! The number is #{random_number}"
end

File: 2__C++_Program/10__number_guessing_game.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;

int main() {
    srand(time(nullptr));

    int choose_number, rand_number;
    int play = 10;
    int won = 0;

    rand_number = (rand() % 100) + 1;

    //PLay the game
    for (int i = 1; i <= play; i++) {
        cout << "[Turn " << i << " / " << play << " ] Choose random number from 0 - 100: ";
        cin >> choose_number;

        if (choose_number > rand_number) {
            cout << "The number is smaller than this!" << endl;
        } else if (choose_number < rand_number) {
            cout << "The number is bigger than this!" << endl;
        } else {
            cout << "You win! The number is " << rand_number << endl;
            won = 1;
            break;
        }
    }

    if (!won) {
        cout << "You lose! The number is " << rand_number << endl;
    }
    
    return 0;
}